### ISSUE TITLE
black screen bug

### ISSUE THREAD

**Issue by judehek** [FFFFFFFXXXXXXX/league_record#31]
I think this has been brought up before at various points, but the app just records a black screen on my laptop. I tried running as admin as well, here are logs from when I ran as admin and quickly joined a practice tool: 
[2024-09-13_13-04.log](https://github.com/user-attachments/files/16998441/2024-09-13_13-04.log)

really appreciate all your hard work on this!

**Comment by FFFFFFFXXXXXXX** [comment]
Did the app work previously? What changed?

From a quick glance at the logs it looks like the recording library tries to use your Intel iGPU instead of your dedicated Nvidia GPU to record the game. You could try telling windows explicitly to use the Nvidia GPU for the app: https://www.majorgeeks.com/content/page/set_preferred_gpu.html

I'll have to take a closer look at what's going on once I'm home.

**Comment by judehek** [comment]
ah sorry, to clarify, it probably never worked on this device, but I've just never tried to play on this laptop.

I set it to use the Nvidia GPU but it still doesn't seem to work (tried with admin again as well):
[2024-09-14_08-24.log](https://github.com/user-attachments/files/17002827/2024-09-14_08-24.log)

**Comment by FFFFFFFXXXXXXX** [comment]
Yeah that makes sense, looks like setting the windows GPU preference doesn't do much either.

Currently the app uses the first GPU available. If you check in the taskmanager that GPU 0 is (most likely) your Intel iGPU. I didn't consider the possibility of someone having more than one GPU.

Sadly I don't think this is fixable on your end.
I'll have to fix this in v2 (unless I find a workaround).

**Comment by judehek** [comment]
no worries!

out of curiousity, is the issue specific to having 2 GPUs or does the recorder not work with integrated graphics? I have access to another laptop that just has integrated graphics, so I might test that

**Comment by FFFFFFFXXXXXXX** [comment]
It should be specific to having 2 GPUs. I have previously tested the app on an old laptop of mine with integrated graphics and it worked.

**Comment by FFFFFFFXXXXXXX** [comment]
I put together a quick proof-of-concept that should address the bug.
You can download the .zip file from this pre-release https://github.com/FFFFFFFXXXXXXX/league_record/releases/tag/test

In this version the logs are in `%LOCALAPPDATA%\fx.LeagueRecord\logs`.

There is now a `"gpuId": 0` setting that *should* allow you to choose which GPU is used to record the game.
`"gpuId": 0` uses the intel GPU and `"gpuId": 1` the NVIDIA GPU.

Would be super awesome if you could test both of those settings and post the logs.

**Comment by judehek** [comment]
just saw it! unfortunately issue still seems to persist.
this is on gpuId 0:
[2024-09-16_11-19.log](https://github.com/user-attachments/files/17017090/2024-09-16_11-19.log)

and here is gpuId 1:
[2024-09-16_11-33.log](https://github.com/user-attachments/files/17017211/2024-09-16_11-33.log)

really appreciate the quick turnaround!

**Comment by FFFFFFFXXXXXXX** [comment]
> Did the app work previously? What changed?
> 
> From a quick glance at the logs it looks like the recording library tries to use your Intel iGPU instead of your dedicated Nvidia GPU to record the game. You could try telling windows explicitly to use the Nvidia GPU for the app: https://www.majorgeeks.com/content/page/set_preferred_gpu.html
> 
> I'll have to take a closer look at what's going on once I'm home.

Could you also try setting an explicit GPU once more (with the old version of the app).
But instead of setting the GPU for `LeagueRecord.exe` set it for `extprocess_recorder.exe` in the `libobs` folder right next to `LeagueRecord.exe`.

**Comment by FFFFFFFXXXXXXX** [comment]
and just to make sure, do you have the LoL window set to borderless mode?

**Comment by judehek** [comment]
it works! interestingly, it only worked when I set it to the Nvidia GPU (high performance) not the integrated one.

I tested with fullscreen and borderless and there didn't seem to be a difference. here are the logs in fullscreen if you're interested:

Nvidia (working):
[2024-09-16_16-05.log](https://github.com/user-attachments/files/17020255/2024-09-16_16-05.log)

Intel (black screen):
[2024-09-16_16-09.log](https://github.com/user-attachments/files/17020257/2024-09-16_16-09.log)

**Comment by FFFFFFFXXXXXXX** [comment]
Nice, for future reference here's an article by the OBS project about this problem: https://obsproject.com/kb/gpu-selection-guide.
